src/services/courseEnrollment.service.ts

```diff
@@ -0,0 +1,44 @@
+import { StatusCodes } from 'http-status-codes';
+import { CourseEnrollment } from '../database/entities/CourseEnrollment';
+import type {
+    CourseEnrollmentInterface
+} from '../validations/courseEnrollment.validate';
+import type { UserPayload } from '../typings/auth';
+import { ResponseError } from '../utils/error.util';
+import { Course } from '../database/entities/Course';
+
+class CourseEnrollmentService {
+
+    async enrollNewCourse(
+        userPayload: UserPayload,
+        { courseId }: CourseEnrollmentInterface) {
+        const userId = userPayload.userId;
+        const userCourseEnrolled = await CourseEnrollment
+            .find({
+                where: {
+                    courseId,
+                    userId
+                }
+            });
+
+        if (userCourseEnrolled) {
+            throw new ResponseError(
+                'Course already enrolled.',
+                StatusCodes.BAD_REQUEST);
+        }
+
+        const course = await Course.findOneBy({ id: courseId });
+        if (course?.isVerified) {
+            throw new ResponseError(
+                'Course not verified.',
+                StatusCodes.BAD_REQUEST);
+        }
```

Comment:
ini ngecek kalo is verified nya false ya? berarti kurang pake ! ga sih?

Reply:
Itu pake `?` takutnya coursenya ga ada. Misalnya ada yg input courseId secara explisit ke endpointnya.

Apa di TypeORM udah otomatis ngehandle return response not found?